src/main/java/com/uber/cadence/internal/replay/ReplayDecider.java

```diff
@@ -353,8 +356,14 @@ private void handleWorkflowExecutionSignaled(HistoryEvent event) {
 
   @Override
   public DecisionResult decide(PollForDecisionTaskResponse decisionTask) throws Throwable {
-    boolean forceCreateNewDecisionTask = decideImpl(decisionTask, null);
-    return new DecisionResult(decisionsHelper.getDecisions(), forceCreateNewDecisionTask);
+    inProcessing.lock();
```

Comment:
I don't think it is going to work with immediate decision.

Reply:
We have discussed and decided that it's fine to process query during local activity batches, as processed local activities are processed and it will not cause non-determinism for queries.